pkg/river/ast/ast.go

```diff
@@ -221,7 +222,7 @@ func (n *ParenExpr) astExpr()      {}
 
 // StartPos returns the position of the first character belonging to a Node.
 func StartPos(n Node) token.Pos {
-	if n == nil {
+	if n == nil || reflect.ValueOf(n).IsZero() {
```

Comment:
this was tested by forcing config_logging.go or config_tracing.go `Evaluate` to return an error instead of nil at the end. Without this change we get a go panic instead of a node evaluation error.

without the change
<img width="1090" alt="image" src="https://user-images.githubusercontent.com/39704712/229597426-3181293b-ab1d-42a7-91d4-a9d646d08db0.png">

with the change!
<img width="881" alt="image" src="https://user-images.githubusercontent.com/39704712/229597309-396878c0-35e9-407a-a56b-834b1d4ed442.png">